rs/rosetta-api/icrc1/rosetta/client/src/lib.rs

```diff
@@ -69,22 +70,46 @@ impl RosettaClient {
             };
 
             // Verify that the signature is correct
-            let verification_key = ic_crypto_ed25519::PublicKey::deserialize_raw(
-                signer_keypair.get_pb_key().as_slice(),
-            )
-            .with_context(|| {
-                format!(
-                    "Failed to convert public key to verification key: {:?}",
-                    signer_keypair.get_pb_key()
-                )
-            })?;
+            match signer_keypair.get_curve_type() {
+                CurveType::Edwards25519 => {
+                    let verification_key = ic_crypto_ed25519::PublicKey::deserialize_raw(
+                        signer_keypair.get_pb_key().as_slice(),
+                    )
+                    .with_context(|| {
+                        format!(
+                            "Failed to convert public key to verification key: {:?}",
+                            signer_keypair.get_pb_key()
+                        )
+                    })?;
+                    if verification_key
+                        .verify_signature(&signable_bytes, signed_bytes.as_slice())
+                        .is_err()
+                    {
+                        bail!("Signature verification failed")
+                    };
+                }
+                CurveType::Secp256K1 => {
+                    let verification_key = ic_crypto_ecdsa_secp256k1::PublicKey::deserialize_sec1(
+                        &signer_keypair.get_pb_key(),
+                    )
+                    .with_context(|| {
+                        format!(
+                            "Failed to convert public key to verification key: {:?}",
+                            signer_keypair.get_pb_key()
+                        )
+                    })?;
+                    if !verification_key
+                        .verify_signature(signable_bytes.as_slice(), signed_bytes.as_slice())
+                    {
+                        bail!("Signature verification failed")
+                    };
+                }
+                _ => bail!(
+                    "Unsupported curve type: {:?}",
+                    signer_keypair.get_curve_type()
+                ),
+            }
```

Comment:
I was going to suggest adding some tests for these error cases, but I couldn't think of how you could reach them?

Reply:
The rosetta client is only for our use really, so would it be safe to say we are only interested in it working as intended for testing?

Reply:
Ah, I see, thanks for the clarification!